brilck.ts

```diff
@@ -318,10 +318,6 @@ function checkOp(env: Env, instr: bril.Operation) {
 }
 
 function checkConst(instr: bril.Constant) {
-  if (!('type' in instr)) {
```

Comment:
Maybe we can just do `(instr as any)`? This is a "defensive programming" kind of trick, where this check is meant to trigger in case the JSON data doesn't match the expected type from our declarations.

Reply:
I've made this change.

The need for this trick kind of surprises me because it sounds like the type is `bril.Constant` when it really should be `bril.Constant | undefined`?(Or the same corresponding argument for any fields that might not exist)

Reply:
Yeah, it's true. The thing is, we want these type definitions to reflect what a "real" Bril program looks like. But at some point, we are doing `JSON.parse` on a Bril program and giving it the type `bril.Program` or whatever—without validating it. Validating the entire program up front would be a huge pain, so this type-cast is a little bit of a lie. But then we can back it up later with some of these ad-hoc checks that second-guess the IR type.